### PR TITLE
Reject modification command if ancestor does not exist or is not in an active state

### LINKED ISSUE #9984: Reject the modification command if the ancestor element instance of an element for activation doesn't exist
## Description

The modification command can contain activation instructions for different elements. If the new element should be activated within a new or a specific flow scope instance (#9646) then the flow scope instance is identified by its element instance key. 

If at least one ancestor element instance key doesn't exist then the command should be rejected. 

This could happen if the key is wrong, or the ancestor element instance is already completed or terminated.

Edge case: the given ancestor element instance is currently in the state `activating` (because an incident occurred during the activation). In this case, the command should be also rejected because the ancestor element instance is not activated yet. But the element instance may be activated eventually.

Blocked by 
- #9646

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

When a modification contains an activate instruction with an ancestor scope key, we must have an element instance for this ancestor in our state. This element instance must be in an active state. If this is not the case we cannot activate an element inside of this scope and the modification should be rejected.

## Related issues

closes #9984 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceRejectionTest.java [hunk 1]
```diff
@@ -434,4 +434,39 @@ public void shouldRejectTerminationOfChildProcess() {
...
+            .expectRejection()
+            .modify();
+
+    // then
+    assertThat(rejection)
+        .describedAs("Expect that the ancestor with key 12345 is not found")
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT)
+        .hasRejectionReason(
+            ("Expected to modify instance of process '%s' but it contains one or more activate"
+                    + " instructions with an an ancestor scope key that does not exist, or is not in an"
```
Comment (korthout) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceRejectionTest.java:468]: > with an an ancestor scope

🔧 double `an` typo

**Review by korthout — APPROVED** [review]
Well done 👏 

👍  Clean and easy-to-follow changes!

🔧 1 tiny typo

LGTM 🚀

File: engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java [hunk 2]
```diff
@@ -280,7 +285,8 @@
~        .flatMap(valid -> validateElementSupported(process, activateInstructions))
         .flatMap(valid -> validateElementInstanceExists(process, terminateInstructions))
         .flatMap(valid -> validateVariableScopeExists(process, activateInstructions))
         .flatMap(valid -> validateVariableScopeIsFlowScope(process, activateInstructions))
+        .flatMap(valid -> validateAncestorExistsAndIsActive(process, value))
```
[the remark this answers is not in the document]
Comment (korthout) [reply engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java:289]: @remcowesterhoud You may need to dismiss this alert

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**